app/controllers/users_controller.rb

```diff
@@ -17,7 +18,12 @@ def create
   end
 
   def show
-    @user = User.find(params[:id])
+    @user = User.find_by_token(params[:id])
```

Comment:
`find_by_token` is very ineffective because it goes through `method_missing`. I suggest using standard `.find_by` - in this case `.find_by(token: params[:id])`.